addons/point_of_sale/models/pos_order.py

```diff
@@ -1533,6 +1533,8 @@ def _compute_total_cost(self, stock_moves):
             product = line.product_id
             if line._is_product_storable_fifo_avco() and stock_moves:
                 product_cost = product._compute_average_price(0, line.qty, line._get_stock_moves_to_consider(stock_moves, product))
+                if (product_cost == 0 and self.order_id.shipping_date and self.refunded_orderline_id):
```

Comment:
Use `float_compare` to check if its 0, like that you take the rounding into account.